If Android, get timezone from system command.

pxx02134 opened the pull request:
In the case of Android, I changed to get timezone from Android's system command 'getprop persist.sys.timezone' if TZ environmental variable not set. It is because Android doesn't seem to have both "/var/db/zoneinfo" and "/etc/timezone" unlike BSD or Linux.

hardening commented:
LGTM, I'm not an Android expert but the provided command seems to provide the same kind of value than the file we're usually scanning

bmiklautz commented:
@freerdp-bot test.

mfleisz commented on libfreerdp/locale/timezone.c:
```diff
@@ -1523,7 +1523,9 @@ char* freerdp_get_unix_timezone_identifier()
 
-#if defined(__FreeBSD__) || defined(__OpenBSD__)
+#if defined(ANDROID)
+	fp = popen("getprop persist.sys.timezone", "r");
```
In case of popen we have to use pclose on Android later on to close the fp correctly.

bmiklautz replied:
as @mfleisz said pclose should later be used if popen was used for opening the i/o stream. Otherwise the behavior is undefined.

akallabeth commented on libfreerdp/locale/timezone.c:
```diff
@@ -1558,7 +1572,11 @@ char* freerdp_get_unix_timezone_identifier()
 
-		fclose(fp);
+#if defined(ANDROID)
+			pclose(fp) ;
```
Indentation wrong

akallabeth commented:
@pxx02134 Could you make the function static (it is only used within the file) and fix the indentation? Except these remarks +1

akallabeth commented:
@pxx02134 What did you do?

pxx02134 commented:
@akallabeth I modified the following
 - created a function to read contents of a time zone file.
 - created a function to open/close a timezone file which is different from a kind of OSs.
 - fixed indentation.

akallabeth commented:
@pxx02134 ah, ok. could you sqash these commits and give them a meaningful message?

akallabeth commented:
@pxx02134 I've taken the liberty to refactor your pull. Could you check it that one works for you?
